internal/v2/command/user.go

```diff
@@ -19,14 +18,27 @@ func (r *CommandSide) AddUser(ctx context.Context, user *usr_model.User) (*usr_m
 		if err != nil {
 			return nil, err
 		}
-		return &usr_model.User{UserName: user.UserName, Human: human}, nil
+		return &domain.User{UserName: user.UserName, Human: human}, nil
 	} else if user.Machine != nil {
 
 	}
 	return nil, caos_errs.ThrowInvalidArgument(nil, "COMMAND-8K0df", "Errors.User.TypeUndefined")
 }
 
-func (r *CommandSide) DeactivateUser(ctx context.Context, userID string) (*usr_model.User, error) {
+func (r *CommandSide) RegisterUser(ctx context.Context, user *domain.User) (*domain.User, error) {
+	if !user.IsValid() {
+		return nil, caos_errs.ThrowPreconditionFailed(nil, "COMMAND-2N9fs", "Errors.User.Invalid")
+	}
+
+	if user.Human != nil {
+
+	} else if user.Machine != nil {
+
+	}
```

Comment:
i believe you're implementing this right now

Reply:
yes